invenio_records_resources/services/files/service.py

```diff
@@ -68,10 +68,11 @@ def _get_record(self, id_, identity, action, file_key=None):
     #
     # High-level API
     #
-    def list_files(self, identity, id_):
+    def list_files(self, identity, id_, with_deleted=False):
```

Comment:
are all the `with_deleted` params being checked against permissions? I guess we need a `<action>_deleted` permission to check if you can perform the action?

Reply:
check how comment here https://github.com/inveniosoftware/invenio-rdm-records/pull/1489/files/492f2201d24152f813a5b8740a6fb64518753717#diff-d79cbaf668935422b2743d81ca9906d9b72ac8b3456c2280ba5a5edacb72796bR387